src/auth/token-provider.ts

```diff
@@ -33,6 +37,11 @@ class DefaultTokenProvider extends TokenProvider {
     private readonly ttl: Duration,
     private readonly tokenUtils: AuthTokens,
   ) {
+    if (ttl.toMillis() > MAX_TOKEN_LIFETIME.toMillis()) {
+      throw new IllegalArgumentError(
+        `Token TTL ${ttl.toHuman()} must be <= ${MAX_TOKEN_LIFETIME.toHuman()}`,
+      );
+    }
     super();
```

Comment:
Question (unrelated and potentially a very dumb one): you're calling `super` here, but this is a base entity. I'm not sure I fully follow, why this works, since it doesn't extend anything. What exactly `super` does in this case?

Reply:
Also, I see that this method returns a `TokenProvider` but there is no return statement. Again, might be missing something here.

Reply:
`DefaultTokenProvider` extends abstract class `TokenProvider`, so `super()` call is required in constructor

Reply:
I'm incredibly dumb, sorry. I thought that this code was related to the TokenProvider, rather than the DefaultTokenProvider (I've skipped 20 lines of code)... Thank you!